### PR TITLE
Xml in jbang

### PR DESCRIPTION
**Pull request by grgrzybek**
# Description

This is first attempt at https://issues.apache.org/jira/browse/CAMEL-18189. `<beans>` element treated simply as root element of _Camel application_ is easy. The `<bean>` element in the Spring sense is hard, because Camel already uses `<bean>` element as bean processor.

However, CAMEL-18189 is about
* root element to collect routes, rests, templates, ...
* add some (to be enhanced later) sort of bean registry population

So I've taken existing `DIRegistry` implementation of `org.apache.camel.support.SupplierRegistry` and populated it by scanning (`org.apache.camel.spi.PackageScanClassResolver`) selected packages.

Here's an example of working XML snippet:

```xml
<camel-app>

    <component-scan base-package="org.apache.camel.dsl.xml.io.beans" />

    <route id="r2">
        <from uri="direct:x2"/>
        <bean ref="bean-from-registry" />
        <to uri="mock:y2"/>
    </route>

</camel-app>
```

where `bean-from-registry` is:

```java
@Named("bean-from-registry")
public class Greeter implements Processor {

    private final GreeterMessage message;

    @Inject
    public Greeter(GreeterMessage message) {
        this.message = message;
    }

    @Override
    public void process(Exchange exchange) throws Exception {
        String msg = exchange.getIn().getBody(String.class);
        exchange.getIn().setBody(message.getMsg() + " " + StringHelper.after(msg, "I'm "));
    }

}
```

And injected `GreeterMessage` is:

```java
@Named
public class GreeterMessage {

    private final String msg = "Hello";

    public String getMsg() {
        return msg;
    }

}
```

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I formatted the code using `mvn -Pformat,fastinstall install && mvn -Psourcecheck`

### CODE REVIEW

**Comment by grgrzybek** [comment]
@davsclaus @oscerd Please have a look - the naming (`use-jsr-330`) is just temporary ;)
If you like the approach, I'll continue.

**Review by oscerd — APPROVED** [review]

**Comment by davsclaus** [comment]
@grgrzybek great work so far.

Yeah we need to think how to make `<bean>` work in `<camel-app>` for declaring beans ala Spring/Blueprint, and then also allow to use <bean> in the Camel route DSL.

Since camel-app is in the model, then it gets exposed to yaml-dsl and other DSLs. Today the yaml-dsl have special "beans" support.

And yeah I dont like the jsr-xxx number - nobody remember these numbers. So we need to come up with a better name. Also I wonder if its needed at all, as it can detect `@Inject` and the Camel annotations as well `@BeanInject` and so on.

**Comment by grgrzybek** [comment]
Sure, as I said - naming is work in progress. Package scanning is a must and we could simply:
* register beans with default ctor
* register beans with `@Named` and `@Inject` constructors

Yaml dsl code is a bit weird for me (at this stage of grasping Camel changes), but definitely there must be one mechanism underneath.

`<camel-app>` and `<beans>` are kind of aliases, but for now the first one looks more natural (as there are no `<bean>` element to justify `<beans>` :D).

Another thought - if Spring's top-level `<beans>` would be supported, it should be able to use `<camelContext>` inside and it's more a work related to CAMEL-19041.

Anyway - the hardest part for me is done (which was to actually write some initial code). I start slowly, but then it's easier for me ;)

File: dsl/camel-yaml-dsl/camel-yaml-dsl-maven-plugin/src/main/java/org/apache/camel/maven/dsl/yaml/GenerateYamlDeserializersMojo.java [hunk 1]
```diff
@@ -847,7 +847,33 @@ private boolean generateSetValue(String modelName, CodeBlock.Builder cb, FieldIn
...
+            Optional<AnnotationValue> adapter = annotationValue(field, XML_JAVA_TYPE_ADAPTER_CLASS, "value");
+            if (adapter.isEmpty()) {
+                return false;
+            }
+            String adapterClass = adapter.get().asClass().name().toString();
+            ClassInfo adapterClassInfo = view.getClassByName(adapter.get().asClass().name());
+            if (adapterClassInfo.superClassType().kind() == Type.Kind.PARAMETERIZED_TYPE) {
+                List<Type> arguments = adapterClassInfo.superClassType().asParameterizedType().arguments();
+                if (arguments.size() == 2) {
+                    // extends XmlAdapter<BeanPropertiesDefinition, Map<String, Object>>
```
Comment (davsclaus) [inline dsl/camel-yaml-dsl/camel-yaml-dsl-maven-plugin/src/main/java/org/apache/camel/maven/dsl/yaml/GenerateYamlDeserializersMojo.java:861]: I guess there is some left over code here that has been disabled

File: tooling/maven/camel-package-maven-plugin/src/main/java/org/apache/camel/maven/packaging/ModelXmlWriterGeneratorMojo.java [hunk 2]
```diff
@@ -779,7 +800,7 @@ private Predicate<Member> accessible(Class<?> clazz) {
         } else if (accessType == XmlAccessType.FIELD) {
             return m -> m.getDeclaringClass() == clazz
                     && ((isSetter(m) || isGetter(m)) && isXmlBindAnnotated(m)
-                            || isField(m) && !Modifier.isStatic(m.getModifiers()) && !Modifier.isTransient(m.getModifiers()));
+                            || isField(m) && /*isNotAnyXml(m) && */!Modifier.isStatic(m.getModifiers()) && !Modifier.isTransient(m.getModifiers()));
```
Comment (davsclaus) [inline tooling/maven/camel-package-maven-plugin/src/main/java/org/apache/camel/maven/packaging/ModelXmlWriterGeneratorMojo.java:803]: I guess a little bit of code that was commented out

File: tooling/maven/camel-package-maven-plugin/src/main/java/org/apache/camel/maven/packaging/ModelXmlWriterGeneratorMojo.java [hunk 3]
```diff
@@ -791,7 +812,13 @@ private Predicate<Member> accessible(Class<?> clazz) {
 
     private boolean isXmlBindAnnotated(Member m) {
         return Stream.of(((AnnotatedElement) m).getAnnotations())
-                .anyMatch(a -> a.getClass().getAnnotatedInterfaces()[0].getType().getTypeName().startsWith("jakarta.xml.bind.annotation."));
+                .anyMatch(a -> a.getClass().getAnnotatedInterfaces()[0].getType().getTypeName().startsWith("jakarta.xml.bind.annotation.")
+                        /*&& !a.getClass().getAnnotatedInterfaces()[0].getType().getTypeName().endsWith("XmlAnyElement")*/);
```
Comment (davsclaus) [inline tooling/maven/camel-package-maven-plugin/src/main/java/org/apache/camel/maven/packaging/ModelXmlWriterGeneratorMojo.java:816]: I guess a little bit of code that was commented out

Comment (grgrzybek) [reply tooling/maven/camel-package-maven-plugin/src/main/java/org/apache/camel/maven/packaging/ModelXmlWriterGeneratorMojo.java]: done. force pushed to PR branch

**Pull request merged**